components/Marker.tsx

```diff
@@ -0,0 +1,58 @@
+import { Image, ImageSourcePropType } from 'react-native';
+import MapView, { Marker as NativeMarker } from 'react-native-maps';
+import { isWeb } from '../utils/platform';
+
+type PinProps = { height: number; width: number; image: ImageSourcePropType };
+
+type MarkerProps = {
+  passthroughKey?: any;
+  coordinate: { latitude: number; longitude: number };
+  height: number;
+  width: number;
+  imageSource: ImageSourcePropType | string;
+  onPress?: () => any;
+  isMapReady?: boolean;
+};
+export const Marker = ({
+  coordinate,
+  height,
+  width,
+  passthroughKey,
+  imageSource,
+  onPress,
+  isMapReady,
+}: MarkerProps) => {
+  // Used only on Web, instead of the Pin component
+  const webIcon = isWeb() && {
+    // @ts-ignore
+    scaledSize: new google.maps.Size(width, height),
+    url: imageSource,
+  };
+
+  return isWeb() ? (
+    // @ts-ignore
+    <MapView.Marker
+      coordinate={coordinate}
+      key={passthroughKey}
+      icon={webIcon}
```

Comment:
minor: the code for webIcon can be inlined and since this is already under the `isWeb` check, you can only inline 
```
icon = {{
    // @ts-ignore
    scaledSize: new google.maps.Size(width, height),
    url: imageSource,
  }}
```